python/pyarrow/array.pxi

```diff
@@ -876,19 +877,6 @@ cdef class Array(_PandasConvertible):
         -------
         diff : str
             A human-readable printout of the differences.
-
-        Examples
-        --------
-        >>> left = pa.array(["one", "two", "three"])
-        >>> right = pa.array(["two", None, "two-and-a-half", "three"])
-        >>> print(left.diff(right))
-
-        @@ -0, +0 @@
-        -"one"
-        @@ -2, +1 @@
-        +null
-        +"two-and-a-half"
```

Comment:
I suppose doctest doesn't like the empty line in the output? In any case, as I commented elsewhere, instead of removing, you can add a doctest skip.